src/main/java/net/dv8tion/jda/core/handle/PresenceUpdateHandler.java

```diff
@@ -177,15 +217,8 @@ protected Long handleInternally(JSONObject content)
                 return null;
 
             //If this was for a Guild, cache it in the Guild for later use in GUILD_MEMBER_ADD
-            if (content.has("guild_id"))
-            {
-                GuildImpl guild = (GuildImpl) api.getGuildById(content.getLong("guild_id"));
+            if (guild != null)
```

Comment:
The removal of the `//cache in relationship stuff` isn't correct here.
We need to be caching this presence for a very soon Group add, similar to how we are doing for the Guild presence caching.

Reply:
I'm not convinced we should be caching relationship presence updates, groups don't send these.